odml/util.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8
+"""
+Module containing general utility functions.
+"""
+
+
+def format_cardinality(in_val):
+    """
+    Checks an input value and formats it towards a custom tuple format
+    used in odml Section, Property and Values cardinality.
+
+    The following cases are supported:
+    (n, n) - default, no restriction
+    (d, n) - minimally d entries, no maximum
+    (n, d) - maximally d entries, no minimum
+    (d, d) - minimally d entries, maximally d entries
+
+    Only positive integers are supported. 'None' is used to denote
+    no restrictions on a maximum or minimum.
+
+    :param in_val: Can either be 'None', a positive integer, which will set
+                   the maximum or an integer 2-tuple of the format '(min, max)'.
+
+    :returns: None or the value as tuple. A ValueError is raised, if the
+              provided value was not in an acceptable format.
+    """
+    exc_msg = "Can only assign positive single int or int-tuples of the format '(min, max)'"
+
+    # Empty values reset the cardinality to None.
+    if not in_val:
+        return None
+
+    # Catch tuple edge cases (0, 0); (None, None); (0, None); (None, 0)
+    if isinstance(in_val, tuple) and len(in_val) > 1 and not in_val[0] and not in_val[1]:
+        return None
+
+    # Providing a single integer sets the maximum value in a tuple.
+    if isinstance(in_val, int) and in_val > 0:
+        return None, in_val
+
+    # Only integer 2-tuples of the format '(min, max)' are supported to set the cardinality
+    if isinstance(in_val, tuple) and len(in_val) == 2:
```

Comment:
do we want to allow lists of two values as well? I think one could easily catch this by 
converting `in_val` to `tuple`